vitrina/orgs/services.py

```diff
@@ -1,16 +1,150 @@
-from vitrina.orgs.models import Representative
+from enum import Enum
+from typing import Type
+
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import Model
+
+from vitrina.datasets.models import Dataset
+from vitrina.orgs.models import Representative, Organization
+from vitrina.requests.models import Request
+from vitrina.resources.models import DatasetDistribution
+from vitrina.users.models import User
 
 
 def has_coordinator_permission(user, organization):
     if user.is_authenticated:
         if user.is_staff:
             return True
-        if user.representative_set.filter(organization=organization).exists():
+        content_type = ContentType.objects.get_for_model(Organization)
+        if user.representative_set.filter(object_id=organization.pk, content_type=content_type).exists():
             representative = (
                 user.
                 representative_set.
-                filter(organization=organization).
+                filter(object_id=organization.pk, content_type=content_type).
                 first()
             )
             return representative.role == Representative.COORDINATOR
     return False
+
+
+class Action(Enum):
+    CREATE = 'create'
+    UPDATE = 'update'
+    DELETE = 'delete'
+
+
+def representative_exists(
+        user: User,
+        content_type: ContentType,
+        object_id: int,
+        role: str = None
+) -> bool:
+    representatives = Representative.objects.filter(
+        user=user,
+        object_id=object_id,
+        content_type=content_type
```

Comment:
```suggestion
        content_object=obj
```

You don't need to pass separately object type and id, just use `content_object` and let Django do the rest. This way `has_perm` function can be cleaned from mangling content types manually.

Reply:
Filtering using content_object doesn't work. According to Django documentation:
"Due to the way [GenericForeignKey](https://docs.djangoproject.com/en/4.1/ref/contrib/contenttypes/#django.contrib.contenttypes.fields.GenericForeignKey) is implemented, you cannot use such fields directly with filters (filter() and exclude(), for example) via the database API. Because a [GenericForeignKey](https://docs.djangoproject.com/en/4.1/ref/contrib/contenttypes/#django.contrib.contenttypes.fields.GenericForeignKey) isn’t a normal field object, these examples will not work"